### PR TITLE
Clarify when the rejectionhandled event does and does not fire

### LINKED ISSUE #22177: rejectionhandled event doesn't fire on every handled rejection, but only if  unhandledrejection events was triggered
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/Window/unhandledrejection_event

### What specific section or headline is this issue about?

https://developer.mozilla.org/en-US/docs/Web/API/Window/unhandledrejection_event#basic_error_logging

### What information was incorrect, unhelpful, or incomplete?

The `rejectionhandled` event doesn't always trigger. Have a look here https://stackoverflow.com/a/57261820/3429430  There should be a note:  

>The [rejectionhandled event](https://developer.mozilla.org/en-US/docs/Web/API/Window/rejectionhandled_event) doesn't fire on every handled rejection, it only fires on rejections that were not handled at first (causing [unhandledrejection events](https://developer.mozilla.org/en-US/docs/Web/API/Window/unhandledrejection_event)) but then did subsequently get a handler attached.

### What did you expect to see?

`rejectionhandled` being fired or an explanation of why id doesn't fire.

### Do you have any supporting links, references, or citations?

https://stackoverflow.com/a/57261820/3429430

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/window/unhandledrejection_event`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Window/unhandledrejection_event
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/window/unhandledrejection_event/index.md
* Last commit: https://github.com/mdn/content/commit/f7dae62645a2c735ed6f6ed63f664bf279fdfc4b
* Document last modified: 2022-09-09T05:19:45.000Z

</details>

### PR DESCRIPTION
**Pull request by sideshowbarker**
Fixes https://github.com/mdn/content/issues/22177. @bergus it would great to have your review on this, if you can make time — since it’s based on your wording from https://stackoverflow.com/a/57261820/441757

### CODE REVIEW

**Comment by bergus** [comment]
I'd recommend to rather change the part "*whenever a JavaScript `Promise` is rejected but after the promise rejection has been handled*". I'd write
```md
The **`rejectionhandled`** event is sent to the script's global scope (usually {{domxref("window")}} but also {{domxref("Worker")}}) whenever a rejected JavaScript {{jsxref("Promise")}} is handled late, after its rejections had caused an {{domxref("Window.unhandledrejection_event", "unhandledrejection")}} event, by subsequently attaching a handler.

This can be used in debugging and for general application resiliency, in tandem with the `unhandledrejection` event, which is sent when a promise is rejected but there is no handler for the rejection at the time.
```

**Comment by sideshowbarker** [comment]
@bergus Thanks much — made it so

File: files/en-us/web/api/window/rejectionhandled_event/index.md [hunk 1]
```diff
@@ -21,9 +21,9 @@ browser-compat: api.Window.rejectionhandled_event
 
 {{APIRef("HTML DOM")}}
 
-The **`rejectionhandled`** event is sent to the script's global scope (usually {{domxref("window")}} but also {{domxref("Worker")}}) whenever a JavaScript {{jsxref("Promise")}} is rejected but after the promise rejection has been handled.
+The **`rejectionhandled`** event is sent to the script's global scope (usually {{domxref("window")}} but also {{domxref("Worker")}}) whenever a rejected JavaScript {{jsxref("Promise")}} is handled late, after its rejections had caused an {{domxref("Window.unhandledrejection_event", "unhandledrejection")}} event, by subsequently attaching a handler.
```
Comment (Josh-Cena) [inline files/en-us/web/api/window/rejectionhandled_event/index.md:24]: The "by subsequently attaching a handler" is a bit hard for me to understand. Does the "handler" mean `onunhandledrejection`, or a `catch()` handler?

Comment (sideshowbarker) [reply files/en-us/web/api/window/rejectionhandled_event/index.md:24]: > The "by subsequently attaching a handler" is a bit hard for me to understand. Does the "handler" mean `onunhandledrejection`, or a `catch()` handler?

@bergus?

Comment (bergus) [reply files/en-us/web/api/window/rejectionhandled_event/index.md:24]: A promise handler, in particular a "rejection handler", but I guess that term won't disambiguate the sentence either. Could be just
```suggestion
The **`rejectionhandled`** event is sent to the script's global scope (usually {{domxref("window")}} but also {{domxref("Worker")}}) whenever a rejected JavaScript {{jsxref("Promise")}} is handled late, after its rejections had caused an {{domxref("Window.unhandledrejection_event", "unhandledrejection")}} event, by subsequently attaching a promise handler.
```
but maybe
```suggestion
The **`rejectionhandled`** event is sent to the script's global scope (usually {{domxref("window")}} but also {{domxref("Worker")}}) whenever a rejected JavaScript {{jsxref("Promise")}} is handled late, i.e. when a handler is attached to it after its rejection had caused an {{domxref("Window.unhandledrejection_event", "unhandledrejection")}} event.
```
```suggestion
The **`rejectionhandled`** event is sent to the script's global scope (usually {{domxref("window")}} but also {{domxref("Worker")}}) whenever a rejected JavaScript {{jsxref("Promise")}} is handled late, i.e. when a handler is attached to the promise after its rejection had caused an {{domxref("Window.unhandledrejection_event", "unhandledrejection")}} event.
```
Notice it's not only `.catch()`, also `.then()` and `.finally()` would attach such a promise handler. I guess the first sentence on the page should be simple, the examples later on the page are hopefully enough to clarify.

**Review by Josh-Cena — APPROVED** [review]
This makes much more sense to me. @bergus This means if `onunhandledrejection` does not attach a handler to the promise, `rejectionhandled` would not fire, right?

**Comment by bergus** [comment]
@Josh-Cena It's usually not the `unhandledrejection` listener that would attach this handler, but some other code that still has a reference to the promise.

**Comment by bsmth** [comment]
Thanks, all. I will merge this shortly 👍🏻

**Pull request merged**